codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/protocol/ServerProtocolTestGenerator.kt

```diff
@@ -549,5 +567,51 @@ class ServerProtocolTestGenerator(
         // These tests are not even attempted to be generated, either because they will not compile
         // or because they are flaky
         private val DisableTests = setOf<String>()
+
+        private fun fixRestJsonSupportsNaNFloatQueryValues(testCase: HttpRequestTestCase): HttpRequestTestCase {
+            // TODO This test does not pass, even after fixing it with this function, because, `NaN` != `NaN`.
+            val params = Node.parse("""{
+                "queryFloat": "NaN",
+                "queryDouble": "NaN",
+                "queryParamsMapOfStringList": {
+                    "Float": ["NaN"],
+                    "Double": ["NaN"]
+                }
+            }""".trimIndent()).asObjectNode().get()
+
+            return testCase.toBuilder().params(params).build()
```

Comment:
I am not sure why are you doing this if it doesn't work. Is the comment outdated?

Reply:
Clarified the comment.